src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -304,6 +311,17 @@ internal void SetCookieOption(Uri uri)
                 }
             }
 
+            private void SetClientCertificateOptions()
+            {
+                if (_clientCertProvider == null)
+                {
+                    return;
+                }
+                SetCurlOption(CURLoption.CURLOPT_SSLCERT, _clientCertProvider._certificateFile);
```

Comment:
> If we are adding a callback, we do not need the file at all. Correct?

Right, in the case of the callback, no file should be necessary.  Haven't had my coffee yet :smile: